### PR TITLE
Genre & Music Classes Implementation

### PR DESCRIPTION
**Pull request by ezekielgn**
In this branch, I worked on the following.

- Create MusicAlbum class.
- Create Genre class with an association to the Item class.
- Make sure that all MusicAlbum class and all Genre class properties are defined and set up in the constructor method.
- Implement methods:
  * Add_item method in the Genre class
  * Should take an instance of the Item class as an input
  * Should add the input item to the collection of items
  * Should add self as a property of the item object (by using the correct setter from the item object)
- Add unit tests for all implemented methods.
- Add options:
  * List all music albums
  * List all genres (e.g 'Comedy', 'Thriller')
  * Add a music album
- Data is preserved by saving collections in .json files.
- Create a schema.sql file with tables that will be analogical to the structure of the classes

### CODE REVIEW

File: genre.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Genre
+  attr_accessor :name
+  attr_reader :id, :items
+
```
Comment (MMGGYY66) [inline genre.rb]: Here you miss add :items in the attr_accessor
and there is no need for the attr_reader

File: genre.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(name)
+    @id = rand(0..1000)
+    @name = name
+    @items = []
+  end
+
```
Comment (MMGGYY66) [inline genre.rb:10]: Well done

File: music_album.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require_relative 'item'
+class MusicAlbum < Item
+  attr_accessor :on_spotify
+
+  def initialize(publish_date, on_spotify)
+    super(publish_date)
+    @on_spotify = on_spotify
+  end
+
+  private
```
Comment (MMGGYY66) [inline music_album.rb]: remove private
as the method needs to be called everywhere

File: app.rb [hunk 4]
```diff
@@ -5,13 +5,19 @@ def initialize()
     @storage = Storage.new('storage')
```
Comment (MMGGYY66) [inline app.rb:5]: you miss to update the app class by adding :genre , :music_albums to the attr_accessor

File: app.rb [hunk 5]
```diff
@@ -5,13 +5,19 @@ def initialize()
...
     @books = []
     @games = []
+    @genres = []
+    @music_albums = []
```
Comment (MMGGYY66) [inline app.rb:9]: well done

File: storage.rb [hunk 6]
```diff
@@ -45,6 +45,29 @@ def retrieve_labels(app)
...
+
+    file.each do |obj|
+      publish_date = obj['publish_date']
+      on_spotify = obj['on_spotify']
+      album = MusicAlbum.new(Date.new(publish_date), on_spotify)
+      app.music_albums << album
+    end
+  end
+
+  def retrive_genres(app)
```
Comment (MMGGYY66) [inline storage.rb:60]: Well done updating storage

File: spec/genre_spec.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
+require_relative 'spec_helper'
+
+describe Genre do
+  context 'Creates a Genre class' do
```
Comment (MMGGYY66) [inline spec/genre_spec.rb:4]: Well done adding unit tests

File: sql/schema.sql [hunk 8]
```diff
@@ -35,4 +35,22 @@ CREATE TABLE game (
...
+  id SERIAL,
+  publish_date date,
+  id_author int,
+  id_genre int,
+  id_label int,
+  on_spotify BOOLEAN,
+  CONSTRAINT fk_author_game FOREIGN KEY (id_author) REFERENCES author (id),
+  CONSTRAINT fk_genre_game FOREIGN KEY (id_genre) REFERENCES genre (id),
+  CONSTRAINT fk_label_game FOREIGN KEY (id_label) REFERENCES label (id)
+  );
```
Comment (MMGGYY66) [inline sql/schema.sql:56]: Well done adding these tables to the schema

**Review by MMGGYY66 — CHANGES_REQUESTED** [review]

File: music_album.rb (hunk 3 above)
Comment (ezekielgn) [reply music_album.rb]: This method actually appears as private in the diagram, and it's only used by the archive public method, so we will need to update this for all classes.

**Pull request merged**